lighthouse-core/lib/i18n/i18n.js

```diff
@@ -130,10 +130,72 @@ function lookupLocale(locale) {
  * @param {Record<string, string | number>} [values]
  */
 function _preprocessMessageValues(icuMessage, values = {}) {
-  const clonedValues = JSON.parse(JSON.stringify(values));
+  let clonedValues = JSON.parse(JSON.stringify(values));
   const parsed = MessageParser.parse(icuMessage);
+
+  const elements = _collectAllCustomElementsFromICU(parsed.elements);
+
+  clonedValues = _processParsedElements(Array.from(elements.values()), clonedValues);
+
+  return clonedValues;
+}
+
+/**
+ * Function to retrieve all 'arguementElement's from an ICU message. An arguementElement
+ * is an ICU element that has an arguement attached like {varName, number, bytes}. This
+ * differs from 'messageElement's which are just arbitrary text in a message.
+ *
+ * Note: This function will recursively inspect plural elements for nested arguementElements.
+ *
+ * @param {Array<import('intl-messageformat-parser').Element>} parsedIcu
+ * @param {Map<string, import('intl-messageformat-parser').Element>} elements
+ */
+function _collectAllCustomElementsFromICU(parsedIcu, elements = new Map()) {
+  // Rescurse into Plurals
+  for (const el of parsedIcu) {
+    if (!el.format || el.format.type !== 'pluralFormat') continue;
+    // We need to find all the elements from the plural format sections, but
+    // they need to be deduplicated. I.e. "=1{hello {icu}} =other{hello {icu}}"
+    // the variable "icu" would appear twice if it wasn't de duplicated.
+    const elemSet = new Map();
+    // Look at all options of the plural (=1{} =other{}...)
+    for (const option of el.format.options) {
+      // Look at each element of each plural option
+      for (const element of option.value.elements) {
```

Comment:
can't this be handled by the recursive step? we shouldn't have a need for two separate loops through `elements` array that adds them to a map